File: Data Structures & Algorithms/count-servers-that-communicate/submission-0.py
class Solution:
    def countServers(self, grid: List[List[int]]) -> int:
        count = 0 
        row_counts = {}
        col_counts = {}
        counted_indices = []
        for i in range (len(grid)):
            for j in range(len(grid[i])):
                if grid[i][j] == 1: 
                    row_counts[i] = 1 + row_counts.get(i, 0) 

        for i in range(0, len(grid)): 
            if row_counts.get(i, 0) > 1: 
                count += row_counts[i]
                for j in range(0, len(grid[i])):
                    if grid[i][j] == 1: 
                        grid[i][j] = '#'
                        counted_indices.append([i, j])

        for i in range (len(grid)):
            for j in range(len(grid[i])):
                if grid[i][j] == 1 or grid[i][j] == '#': 
                    col_counts[j] = 1 + col_counts.get(j, 0)

        for i in range (len(grid)):
            for j in range(len(grid[i])):
                if col_counts.get(j, 0) > 1 and grid[i][j] == 1: 
                    count += 1 

        return count 

File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        int length = s.size();
        if (length != t.size()) return false;
        unordered_map<char, int> umap;

        for (int i = 0; i < length; i++) {
            if (umap.count(s.at(i))) umap[s.at(i)] += 1;
            else umap[s.at(i)] = 1; 
        }
        for (int i = 0; i < length; i++) {
            if (!umap.count(t.at(i))) return false;
            else {
                if (umap[t.at(i)] < 1) return false;
                umap[t.at(i)] -= 1;
            }
        }
        return true;
    }
};
